### ISSUE TITLE
fastText and word2vec file formats incompatible?

### ISSUE THREAD

**Issue by g-karthik** [facebookresearch/fastText#271]
I downloaded the pre-trained English Wikipedia vectors file (`wiki.en.vec`), and I tried to compute the syntactic and semantic analogy task accuracies as described in the first of Mikolov's word2vec papers as follows:

1) I built the word2vec repository by simply doing `make`.
2) I ran `./compute-accuracy wiki.en.vec 0 < questions-words.txt`, i.e., I pass the pre-trained vectors file to the `compute-accuracy` binary from word2vec along with a threshold of 0 in order to consider the entire vocabulary instead of by default restricting it to 30000, and I also send in the accuracy computation dataset `questions-words.txt` using `<` because I noticed that the code reads the dataset from `stdin`.

In response, I simply get the following bunch of NaNs (this doesn't change even if I change the threshold value to 30000 or anything else):

capital-common-countries:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
capital-world:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
currency:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
city-in-state:
^[[AACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
family:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram1-adjective-to-adverb:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram2-opposite:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram3-comparative:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram4-superlative:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram5-present-participle:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram6-nationality-adjective:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram7-past-tense:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram8-plural:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
gram9-plural-verbs:
ACCURACY TOP1: 0.00 %  (0 / 1)
Total accuracy: -nan %   Semantic accuracy: -nan %   Syntactic accuracy: -nan % 
Questions seen / total: 0 19544   0.00 % 

Can someone please explain why the English pre-trained vectors don't seem to work with word2vec's accuracy computation code? I took a look at `compute-accuracy.c` and it does look like it expects standard vector file formatting convention and I took a look at `wiki.en.vec` as well, and it does look like it is formatted in standard convention.

Also, in the fastText paper, word analogy accuracies with fastText vectors are presented and the paper cites Mikolov's word2vec paper there -- clearly, the same dataset was used, and presumably the same word2vec `compute-accuracy.c` file was used to obtain the presented numbers. So could someone please explain what's going wrong?

**Comment by pbamotra** [comment]
These might be helpful for your case:

https://radimrehurek.com/gensim/models/keyedvectors.html
https://radimrehurek.com/gensim/models/wrappers/fasttext.html
